crates/stages/api/src/pipeline/mod.rs

```diff
@@ -280,6 +280,51 @@ impl<N: ProviderNodeTypes> Pipeline<N> {
         to: BlockNumber,
         bad_block: Option<BlockNumber>,
     ) -> Result<(), PipelineError> {
+        // Add validation before starting unwind
+        let provider = self.provider_factory.provider()?;
+        let latest_block = provider.last_block_number()?;
+
+        // Get the actual pruning configuration
+        let prune_modes = provider.prune_modes_ref();
+
+        // Check account history pruning
+        if let Some(PruneMode::Distance(history_limit)) = prune_modes.account_history {
+            if latest_block.saturating_sub(to) > history_limit {
```

Comment:
as @Rjected requested, I'd like to convert this to a function of prune_modes that returns a bool if this is the case